pkg/resource/snapshot/custom_set_output.go

```diff
@@ -0,0 +1,90 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package snapshot
+
+import (
+	svcapitypes "github.com/aws-controllers-k8s/memorydb-controller/apis/v1alpha1"
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	"github.com/aws/aws-sdk-go/service/memorydb"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func (rm *resourceManager) CustomDescribeSnapshotSetOutput(
+	resp *memorydb.DescribeSnapshotsOutput,
+	ko *svcapitypes.Snapshot,
+) (*svcapitypes.Snapshot, error) {
+	if len(resp.Snapshots) == 0 {
+		return ko, nil
+	}
+	elem := resp.Snapshots[0]
+	rm.customSetOutput(elem, ko)
+	return ko, nil
+}
+
+func (rm *resourceManager) CustomCreateSnapshotSetOutput(
+	resp *memorydb.CreateSnapshotOutput,
+	ko *svcapitypes.Snapshot,
+) (*svcapitypes.Snapshot, error) {
+	rm.customSetOutput(resp.Snapshot, ko)
+	return ko, nil
+}
+
+func (rm *resourceManager) CustomCopySnapshotSetOutput(
+	resp *memorydb.CopySnapshotOutput,
+	ko *svcapitypes.Snapshot,
+) *svcapitypes.Snapshot {
+	rm.customSetOutput(resp.Snapshot, ko)
```

Comment:
Why is this necessary for create? (Ignoring conditions) Does `KMSKeyID` not get populated correctly from the response?